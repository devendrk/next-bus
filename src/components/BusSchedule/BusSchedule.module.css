.cardBody {
  display: flex;
  justify-content: space-between;
  text-align: left;
  padding: 0 10%;
  line-height: 2;
  font-size: large;
  border-top: 1px solid rgb(219, 216, 216);
  font-weight: 500;
}
