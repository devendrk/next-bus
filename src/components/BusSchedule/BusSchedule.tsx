import React from "react";

import { convertTimeFormat } from "../../utils/index";
import styles from "./BusSchedule.module.css";
interface Props {
  nextBuses?: [] | any;
}

const BusSchedule: React.FC<Props> = ({ nextBuses }) => {
  console.log("nb", nextBuses);
  return (
    <div>
      {nextBuses.map((nb: any) => (
        <div
          className={styles.cardBody}
          key={nb.headsign.length * nb.scheduledArrival}
        >
          <p>{nb.headsign}</p>
          <p>{convertTimeFormat(nb.scheduledArrival)}</p>
        </div>
      ))}
    </div>
  );
};

export default BusSchedule;
