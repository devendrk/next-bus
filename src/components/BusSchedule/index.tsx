import React from "react";

import { convertTimeFormat } from "../../utils/index";
import styles from "./BusSchedule.module.css";

import { Stops } from "../../types";
interface Props {
  nextBuses: Stops[];
}

const BusSchedule: React.FC<Props> = ({ nextBuses }) => {
  return (
    <div>
      {nextBuses.map((nb: any, index: number) => (
        <div className={styles.cardBody} key={index}>
          <p>{nb.trip.route.shortName}</p>
          <p className="short-name">{nb.headsign}</p>
          <p>{convertTimeFormat(nb.scheduledArrival)}</p>
        </div>
      ))}
    </div>
  );
};

export default BusSchedule;
