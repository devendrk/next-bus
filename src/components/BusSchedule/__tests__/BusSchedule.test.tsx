import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import BusSchedule from "..";

describe("<Card />", () => {
  const nextBuses = [
    {
      headsign: "Helsinki",
      scheduledArrival: 4200,
      trip: { route: { shortName: "abc", id: "abc" } },
    },
  ];
  it("BusSchedule", () => {
    const { container } = render(<BusSchedule nextBuses={nextBuses} />);
    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
