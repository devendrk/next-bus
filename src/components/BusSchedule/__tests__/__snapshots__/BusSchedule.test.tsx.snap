// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> BusSchedule 1`] = `
<div>
  <div>
    <div
      class="cardBody"
    >
      <p>
        abc
      </p>
      <p
        class="short-name"
      >
        Helsinki
      </p>
      <p>
        01:10
      </p>
    </div>
  </div>
</div>
`;
